### PR TITLE
Adding NuGet naming conventions

### PR DESCRIPTION
**Pull request by MichielOda**

### CODE REVIEW

**Comment by janstaelensskyline** [comment]
* Perhaps change the name of the file to : Publishing to the public NuGet.org store.
* I think we should make SDK Style projects mandatory for libraries so we can avoid problems with commandlines not working (nuget versus dotnet). Now that we already have most of the tools necessary to workaround SDK style problems and everything github will be SDK.  @PedroDebevere  what is your thoughts here? 
*

File: develop/TOOLS/TOONuGet/NuGet_Naming_Conventions.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        - Skyline.DataMiner.CICD.DMApp.XXX (specific code to convert a XXX solution/repository to dmapp package)
+
+## Mandatory metadata
+
+```xml
+<PropertyGroup>
+    ...
+    <Authors>SkylineCommunications</Authors>
+    <Company>Skyline Communications</Company>
+    <Descrption>{Short description about the NuGet}</Description>
```
Comment (janstaelensskyline) [inline develop/TOOLS/TOONuGet/NuGet_Naming_Conventions.md:45]: I would add here that if there is a specific line of code as an "entry point" it should be added here. This makes it much more easy to start using any library.  {Code Entry Point: line of code that gets a developer started with your library}
Example:
Library facilitating communication between DataMiner elements and automationscripts by providing a Command Pattern Framework. Code Entry Point: IInterAppCall command = InterAppCallFactory.CreateNew();

File: develop/TOOLS/TOONuGet/NuGet_Naming_Conventions.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    - In general .NET Standard 2.0 is a good choice as that gives the most flexibility to be used somewhere else.
+- Try to make use of the SDK project style. This is more future proof and a lot cleaner.
+    - .NET Framework can still be used, but that is easily done by changing the value to 'net462'.
+    ```xml
+    <PropertyGroup>
+        ...
+        <TargetFramework>netstandard2.0</TargetFramework> => <TargetFramework>net462</TargetFramework>
+        ...
+    </PropertyGroup>
+    ```
```
Comment (janstaelensskyline) [inline develop/TOOLS/TOONuGet/NuGet_Naming_Conventions.md:96]: Maybe add a part about adding a README.md file to the projects as well.
We could reference to the README.md of DataMinerSystem.Automation as an example perhaps.

**Review by janstaelensskyline — COMMENTED** [review]
A few extra comments

File: develop/TOOLS/TOONuGet/NuGet_Naming_Conventions.md (hunk 1 above)
Comment (janstaelensskyline) [reply develop/TOOLS/TOONuGet/NuGet_Naming_Conventions.md:45]: Actually: I also added [Minimum Required DataMiner: 10.1.0.6]  to the start of the description for the Class Library things. So it's clear.

<Description>{[Minimum Required DataMiner: A.B.C.D} {Short description about the NuGet} {Code Entry Point: line of code that gets a developer started with your library}</Description>

The description is the first thing a user generally see's in the NuGetManager.
Also you misspelled the first opening tag: Descrption   should be Description

File: develop/toc.yml [hunk 3]
```diff
@@ -5261,6 +5261,9 @@
           topicUid: Consuming_NuGet
         - name: Producing NuGet packages
           topicUid: Producing_NuGet
+          items:
+            - name: Nuget Naming Conventions
```
Comment (PedroDebevere) [inline develop/toc.yml:5265]: Change casing to NuGet naming conventions

File: develop/TOOLS/TOONuGet/NuGet_Naming_Conventions.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+- NuGet must include a license file when published to nuget.org
+- NuGet must be signed otherwise this will be blocked by nuget.org
+    - When creating the NuGet via [SLC SE Repository Manager](xref:TOOSLCSERepositoryManager), the pipeline will take care of the signing and publishing.
+- [Mandatory metadata](#mandatory-metadata)
+
+<br/>
+
+- If there is a dependency on a DataMiner DLL (from the Skyline DataMiner/Files folder), use the corresponding NuGet (Skyline.DataMiner.Files.XXX).
+    - If it is not there, contact the [Data-Acquisition](mailto:support.data-acquisition@skyline.be) domain. They can then include this DLL in the automatic NuGet creation from DataMiner versions.
+- If there is a dependency on multiple DataMiner DLLs that are default included in a QAction/Exe, then make use of the DevPacks (Skyline.DataMiner.Dev.XXX)
```
Comment (PedroDebevere) [inline develop/TOOLS/TOONuGet/NuGet_Naming_Conventions.md:24]: DLLs that are default included => DLLs that are by default included

File: develop/TOOLS/TOONuGet/NuGet_Naming_Conventions.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+<br/>
+
+- Use the syntax: 'Skyline.DataMiner.\[Categories\].\[Name\]'. Optionally at the end you could have '.Protocol', '.Automation', '.Common', ... to specify on where it can be used (usage of SLManagedScripting, SLManagedAutomation, ...)
```
Comment (PedroDebevere) [inline develop/TOOLS/TOONuGet/NuGet_Naming_Conventions.md:28]: - Use the syntax: 'Skyline.DataMiner.\[Categories\].\[Name\]'. Optionally at the end you could have '.Protocol', '.Automation', '.Common', ... to specify on where it can be used (usage of SLManagedScripting, SLManagedAutomation, ...)
 => 
- Use the syntax: 'Skyline.DataMiner.<Category>.<Name>'. Optionally at the end you could have '.Protocol', '.Automation', '.Common', ... to specify where it can be used (usage of SLManagedScripting, SLManagedAutomation, ...)

File: develop/TOOLS/TOONuGet/NuGet_Naming_Conventions.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    - Skyline.DataMiner.\[Categories\].\[Name\].Common (Shared code that can run either in Script or Protocol. Communicates with DataMiner via SLNet or [DataMinerSystem](https://www.nuget.org/packages?q=Skyline.DataMiner.Core.DataMinerSystem))
```
Comment (PedroDebevere) [inline develop/TOOLS/TOONuGet/NuGet_Naming_Conventions.md:29]: same remark as above, replace [] with <>, same for lines below

File: develop/TOOLS/TOONuGet/NuGet_Naming_Conventions.md [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+
+## Mandatory rules
+
+- NuGet must have a name that starts with 'Skyline.DataMiner'.
+- NuGet must include a license file when published to nuget.org
+- NuGet must be signed otherwise this will be blocked by nuget.org
+    - When creating the NuGet via [SLC SE Repository Manager](xref:TOOSLCSERepositoryManager), the pipeline will take care of the signing and publishing.
+- [Mandatory metadata](#mandatory-metadata)
+
+<br/>
```
Comment (PedroDebevere) [inline develop/TOOLS/TOONuGet/NuGet_Naming_Conventions.md:20]: remove this html

File: develop/TOOLS/TOONuGet/NuGet_Naming_Conventions.md [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+
+- If there is a dependency on a DataMiner DLL (from the Skyline DataMiner/Files folder), use the corresponding NuGet (Skyline.DataMiner.Files.XXX).
+    - If it is not there, contact the [Data-Acquisition](mailto:support.data-acquisition@skyline.be) domain. They can then include this DLL in the automatic NuGet creation from DataMiner versions.
+- If there is a dependency on multiple DataMiner DLLs that are default included in a QAction/Exe, then make use of the DevPacks (Skyline.DataMiner.Dev.XXX)
+
+<br/>
```
Comment (PedroDebevere) [inline develop/TOOLS/TOONuGet/NuGet_Naming_Conventions.md:26]: remove this html

File: develop/TOOLS/TOONuGet/NuGet_Naming_Conventions.md (hunk 1 above)
Comment (PedroDebevere) [reply develop/TOOLS/TOONuGet/NuGet_Naming_Conventions.md:45]: Not sure if the entry point should be part of the description. I feel it belongs more to the readme file.

File: develop/TOOLS/TOONuGet/NuGet_Naming_Conventions.md [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+    <Descrption>{Short description about the NuGet}</Description>
+    <PackageTags>Skyline;DataMiner</PackageTags>
+    <PackageProjectUrl>https://skyline.be</PackageProjectUrl>
+    <PackageReadmeFile>README.md</PackageReadmeFile>
+    ...
+</PropertyGroup>
+```
+
+- Authors has to be without whitespace.
+- PackageTags need to have at least 'Skyline' and 'DataMiner'. Extra tags can be added in wanted.
```
Comment (PedroDebevere) [inline develop/TOOLS/TOONuGet/NuGet_Naming_Conventions.md:54]: in wanted => if desired.

File: develop/TOOLS/TOONuGet/NuGet_Naming_Conventions.md [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+<PropertyGroup>
+    ...
+    <PackageLicenseFile>LICENSE.txt</PackageLicenseFile>
+    <PackageRequireLicenseAcceptance>True</PackageRequireLicenseAcceptance>
+    <PackageIcon>Icon.png</PackageIcon>
+    ...
+</PropertyGroup>
+```
+
+## Restricted names
```
Comment (PedroDebevere) [inline develop/TOOLS/TOONuGet/NuGet_Naming_Conventions.md:69]: Reserverd names

File: develop/TOOLS/TOONuGet/NuGet_Naming_Conventions.md [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+
+- Skyline.DataMiner.Files.XXX (DataMiner DLLs - Autogenerated by tool from Data-Acquisition domain)
+- Skyline.DataMiner.Dev.XXX (DevPacks - Autogenerated by tool from Data-Acquisition domain)
+- Skyline.DataMiner.CICD.XXX (Managed by Data-Acquisition domain)
+
+## Names to be used
+
+- Skyline.DataMiner.Core.XXX
+    - Holds the items that have a direct link to DataMiner (AppPackage creation/installation, DataMinerSystem, ...)
+    - DxM will also fall under this. Take in account the have clear error handling in case the DxM is optional and not available.
```
Comment (PedroDebevere) [inline develop/TOOLS/TOONuGet/NuGet_Naming_Conventions.md:79]: Take in account the have clear error handling in case the DxM is optional and not available. => this remark feels a bit out of scope here.

File: develop/TOOLS/TOONuGet/NuGet_Naming_Conventions.md [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+- Skyline.DataMiner.Utils.XXX
+    - Holds all the utilities like helpers, parsers, ...
+        - E.g.: SNMP trap parsing, rate calculations, table context menu, ...
+
+## Suggestions and reminders
+
+- Take in account the target framework. If you have a dependency to .NET Framework 4.6.2 (e.g.: some DataMiner DLLS) then you need to target that as well.
```
Comment (PedroDebevere) [inline develop/TOOLS/TOONuGet/NuGet_Naming_Conventions.md:86]: DLLS => DLLs

File: develop/TOOLS/TOONuGet/NuGet_Naming_Conventions.md [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+    - In general .NET Standard 2.0 is a good choice as that gives the most flexibility to be used somewhere else.
+- Try to make use of the SDK project style. This is more future proof and a lot cleaner.
+    - .NET Framework can still be used, but that is easily done by changing the value to 'net462'.
```
Comment (PedroDebevere) [inline develop/TOOLS/TOONuGet/NuGet_Naming_Conventions.md:89]: Using SDK-style projects, you can still target .NET Framework 4.6.2:

File: develop/TOOLS/TOONuGet/NuGet_Naming_Conventions.md (hunk 1 above)
Comment (PedroDebevere) [reply develop/TOOLS/TOONuGet/NuGet_Naming_Conventions.md:45]: @janstaelensskyline : I'm also in favour of using SDK style projects by default

File: develop/TOOLS/TOONuGet/Producing_NuGet.md [hunk 14]
```diff
@@ -4,80 +4,153 @@ uid: Producing_NuGet
...
+
+## General
+
+Below rules are mainly focused for NuGets that are published to nuget.org. It is highly adviced to use the same rules for internal NuGets as well as that would make it easy transferable to nuget.org when necessary.
+
+### Mandatory rules for nuget.org
 
-To configure the name and version of your NuGet packages, all standard options are available. The easiest way to do this is to go to the project settings and adjust the assembly info.
+- NuGet needs to include a license file
+- NuGet needs to be signed as it will not be accepted
```
Comment (PedroDebevere) [inline develop/TOOLS/TOONuGet/Producing_NuGet.md:24]: NuGet needs to be signed as it will not be accepted => NuGet needs to be signed as only signed packages can be published on nuget.org

File: develop/TOOLS/TOONuGet/Producing_NuGet.md [hunk 15]
```diff
@@ -4,80 +4,153 @@ uid: Producing_NuGet
...
+> [!NOTE]
+> Prior to DataMiner 10.1.11 (RN 30755) when using a .NET Standard 2.0 NuGet in a QAction or Exe, you need to manually add a reference to .NET Standard.
+>
+> ```xml
+> <ItemGroup>
+>   <Reference Include="netstandard" />
+> </ItemGroup>
+> ```
+>
+> When targeting .NET Framework 4.6.2 you'll get a warning icon, but this can be ignored.
```
Comment (PedroDebevere) [inline develop/TOOLS/TOONuGet/Producing_NuGet.md:62]: TBD

File: develop/TOOLS/TOONuGet/Producing_NuGet.md [hunk 16]
```diff
@@ -4,80 +4,153 @@ uid: Producing_NuGet
...
 
-## Manual NuSpec file creation
+## Naming conventions
 
-Alternatively, you can create NuSpec files yourself and place these next to the projects. This can be useful where the automatically created NuSpec file does not meet your needs (e.g. in case you want to provide some tags or customize the content of the package).
+### Mandatory rules
 
-The pipeline will perform the following checks on the NuSpec file:
+- NuGet must have a name that starts with `Skyline.DataMiner`.
```
Comment (PedroDebevere) [inline develop/TOOLS/TOONuGet/Producing_NuGet.md:68]: NuGet must have a name that starts with `Skyline.DataMiner.`.

File: develop/TOOLS/TOONuGet/Producing_NuGet.md [hunk 17]
```diff
@@ -4,80 +4,153 @@ uid: Producing_NuGet
...
-To configure the name and version of your NuGet packages, all standard options are available. The easiest way to do this is to go to the project settings and adjust the assembly info.
+- NuGet needs to include a license file
+- NuGet needs to be signed as only signed packages can be published on nuget.org
+  - When creating the NuGet via [SLC SE Repository Manager](xref:TOOSLCSERepositoryManager), the pipeline will take care of signing and publishing.
 
-For example:
+### Project type
 
-![Assembly info in project settings](~/develop/images/Assembly_info_NuGet.png)
+Make use of the SDK project style. This gives the advantage of using dotnet tools in the pipeline and has the latest features. It also cleans up the csproj file which reduces potential merge conflicts.
```
Comment (janstaelensskyline) [inline develop/TOOLS/TOONuGet/Producing_NuGet.md:30]: When using SDK project style, open your repository jenkins file with a text editor and make sure the master file being used is the 'MasterFiles\\Custom Solution\\JenkinsSDK'.

File: develop/TOOLS/TOONuGet/Producing_NuGet.md [hunk 18]
```diff
@@ -4,80 +4,153 @@ uid: Producing_NuGet
...
-If no NuSpec file was found, the pipeline will automatically create one by extracting the required metadata from the AssemblyInfo.cs file of the project.
+If there is a dependency on multiple DataMiner DLLs that are by default included in a QAction/Exe, then make use of the DevPacks ([Skyline.DataMiner.Dev.XXX](https://www.nuget.org/packages?q=Skyline.DataMiner.Dev.))
 
-The pipeline expects the following to present in the AssemblyInfo.cs file:
+### Target Framework
 
-- The name of the assembly must start with `Skyline.DataMiner.`.
-- A description must be provided (with a maximum length of 4000 characters).
-- A copyright must be provided.
+If possible, try to target .NET Standard 2.0 as that gives the most flexibility to be used somewhere else.
```
Comment (janstaelensskyline) [inline develop/TOOLS/TOONuGet/Producing_NuGet.md:49]: This is incorrect. We recently learned that NET Standard 2.0 doesn't actually work well unless you're using the very latest feature releases. I think we should continue making people target .NET Framework 4.6.2 for now.

Comment (MichielOda) [reply develop/TOOLS/TOONuGet/Producing_NuGet.md:52]: As discussed internally, targeting .NET 2.0 is fine. To make use of a .NET Standard NuGet in a QAction or Exe, you have to add a reference to netstandard. This is documented a bit below with the Target Framework section. I've changed it from a Note to Important to highlight it more.

**Comment by MariekeGO** [comment]
@MichielOda, could you double-check if everything is still OK after my review?

**Comment by MichielOda** [comment]
> @MichielOda, could you double-check if everything is still OK after my review?

image
Not sure about the comma? As the sentences shouldn't be splitted up there.

**Comment by MariekeGO** [comment]
> > @MichielOda, could you double-check if everything is still OK after my review?
> 
> image Not sure about the comma? As the sentence shouldn't be splitted up there.

Corrected!

**Comment by MichielOda** [comment]
> > > @MichielOda, could you double-check if everything is still OK after my review?
> > 
> > 
> > image Not sure about the comma? As the sentence shouldn't be splitted up there.
> 
> Corrected!

image
Euhm, I guess something went wrong? Cause the comma is still there?

**Comment by MichielOda** [comment]
Looks good to me! Thanks for the improvements!

**Review by MariekeGO — APPROVED** [review]
